plugin/src/lib/utilities/richTextConversion.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  slateToHtml,
+  payloadSlateToHtmlConfig,
+  type SlateToHtmlConfig,
+  htmlToSlate,
+  payloadHtmlToSlateConfig,
+  type HtmlToSlateConfig,
+} from "@slate-serializers/html"
+import type { Descendant } from "slate";
+
+import type { SerializedEditorState } from 'lexical'
+import {
+  BlockNode,
+  HTMLConverter,
+  type SanitizedEditorConfig,
+  convertLexicalToHTML,
+  consolidateHTMLConverters,
+} from '@payloadcms/richtext-lexical'
+
+import { createHeadlessEditor } from '@lexical/headless';
+import {$generateNodesFromDOM} from '@lexical/html';
+import { $getRoot, $getSelection } from 'lexical'
+import { Window } from 'happy-dom'
+
+const BlockHTMLConverter: HTMLConverter<any> = {
+  converter: async ({ node }) => {
+    return `<span data-block-id=${node.fields.id}}></span>`
+  },
+  nodeTypes: [BlockNode.getType()],
+}
+
+export const convertLexicalToHtml = async (editorData: SerializedEditorState, editorConfig: SanitizedEditorConfig) => {
+  return await convertLexicalToHTML({
+    converters: [
+      ...consolidateHTMLConverters({ editorConfig }),
+      BlockHTMLConverter,
+    ],
+    data: editorData,
+  })
+}
+
+export const convertHtmlToLexical = async (htmlString: string, editorConfig: SanitizedEditorConfig) => {
+  const headlessEditor = createHeadlessEditor({ ...editorConfig, nodes: [] });
+  headlessEditor.update(() => {
+    
+    const contextWindow = ((): Window => {
+      if (process.env['NODE_ENV'] === "test") {
+        
+        return new Window();
+      }
+      /* eslint-disable  @typescript-eslint/no-explicit-any */
+      return (window as any) as Window
+    })()
```

Comment:
This is tricky. Tests run in Node.js. Webpack compiles Payload CMS for the browser.

If no condition is added here, tests will pass but Webpack will not compile.

Tried https://github.com/capricorn86/happy-dom-without-node but in this case, we face the problem of importing an ESM module into a commonjs project. [Payload 3 moves to ESM](https://github.com/payloadcms/payload/discussions/3802) but for now, this project supports Payload 2.

Functionality is untested in production because the function differs in the test environment. This should be resolved when Payload is upgraded.

Finally, some commented out code to demonstrate the attempts to use different parsers. Ideally we'd be able to use something like `htmlparser2` but it does not produce a DOM object with the APIs expected by Lexical in https://github.com/facebook/lexical/blob/181404301abaa51c4d82d14d62f0312e74d323a9/packages/lexical-html/src/index.ts#L41.

```
// import * as parse5 from "parse5";
// import * as htmlparser2 from "htmlparser2";
// import { getChildren, getName } from "domutils";
// import { parseFromString } from 'dom-parser';
// import { DOMParser } from'@xmldom/xmldom'
// import { parse, Node } from 'node-html-parser';

// In a headless environment you can use a package such as JSDom to parse the HTML string.
// const document = htmlparser2.parseDocument(htmlString);

//const document = new DOMParser().parseFromString(`<html><body>${htmlString}</body></html>`, 'text/html')

/* eslint-disable  @typescript-eslint/no-explicit-any */

// const document = parse(htmlString)

    /**
(document as any).body = {
  childNodes: getChildren(document).map(childNode => ({
    ...childNode,
    nodeName: getName(childNode as any),
  }))
}
*/
```